src/plugins/wizard/public/application/components/data_tab/secondary_panel.tsx

```diff
@@ -46,9 +49,26 @@ export function SecondaryPanel() {
   const showAggParamEditor = !!(aggConfig && indexPattern);
 
   const closeMenu = useCallback(() => {
-    // Save the agg if valid else discard
-    dispatch(saveAgg(valid));
-  }, [dispatch, valid]);
+    dispatch(editDraftAgg(undefined));
+  }, [dispatch]);
+
+  const handleSetValid = useCallback(
+    (isValid: boolean) => {
+      // Set validity state globally
+      dispatch(
+        setValid({
+          key: EDITOR_KEY,
+          valid: isValid,
+        })
+      );
+
+      // Autosave changes if valid
+      if (valid) {
+        dispatch(saveDraftAgg());
+      }
+    },
+    [dispatch, valid]
+  );
```

Comment:
Are `isValid` and `valid` supposed to be different? If yes, the naming can improve in a future change.

Reply:
Yeah, is `isValid` is the value passed to the callback function. `valid` is the value stored in state. Yeah I agree, naming can improve. Will take note to update it in the next PR :)